offer-manager-domain/src/main/groovy/life/qbic/business/offers/Offer.groovy

```diff
@@ -261,28 +261,35 @@ class Offer {
      * @return The net offer price including overheads
      */
     double getTotalNetPriceWithOverheads() {
-        return calculateNetPrice() - getTotalDiscountAmount() + getOverheadSum()
+        return (calculateNetPrice() + getOverheadSum()) as BigDecimal
     }
 
     /**
      * The overhead price amount of all service items without VAT.
      *
      * @return The calculated overhead amount of the selected items.
      */
-    double getOverheadSum() {
+    BigDecimal getOverheadSum() {
         if (items.empty) {
             return 0
         }
-        return items.sum {calculateItemOverhead(it)} as double
+        return items.sum {calculateItemOverhead(it)} as BigDecimal
     }
 
-    private BigDecimal calculateItemNet(ProductItem item) {
+    private BigDecimal determineUnitPrice(ProductItem item) {
         BigDecimal unitPrice = selectedCustomerAffiliation.category == AffiliationCategory.INTERNAL ? item.product.internalUnitPrice : item.product.externalUnitPrice
-        return unitPrice * item.quantity
+        return unitPrice
+    }
+
+    private BigDecimal calculateItemNet(ProductItem item) {
+        BigDecimal unitPrice = determineUnitPrice(item)
+        BigDecimal totalItemPrice = unitPrice * item.quantity
+        return totalItemPrice - Math.max(discountAmountForProductItem(item),
+                storageDiscountAmountForProductItem(item))
     }
 
-    private double calculateItemOverhead(ProductItem item) {
-        return (calculateItemNet(item) - item.quantityDiscount) * overhead
+    private BigDecimal calculateItemOverhead(ProductItem item) {
```

Comment:
Why was the discount removed here again? I thought that this was specifically requested in the past.

Reply:
Because the discount is already considered in the net sum.